UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Services/ProductDataService.cs

```diff
@@ -1,128 +1,147 @@
+            IProductDataSinceDateTimeValidator productDataSinceDateTimeValidator,
+            ISalesCatalogueService salesCatalougeService,
+            IMapper mapper)
         {
             this.productIdentifierValidator = productIdentifierValidator;
             this.productVersionsValidator = productVersionsValidator;
             this.productDataSinceDateTimeValidator = productDataSinceDateTimeValidator;
+            this.salesCatalogueService = salesCatalougeService;
+            this.mapper = mapper;
         }
 
-        public async Task<ExchangeSetResponse> CreateProductDataByProductIdentifiers(ProductIdentifierRequest productIdentifierRequest)
+        public async Task<ExchangeSetServiceResponse> CreateProductDataByProductIdentifiers(ProductIdentifierRequest productIdentifierRequest)
         {
-            LinkSetBatchStatusUri linkSetBatchStatusUri = new LinkSetBatchStatusUri()
+            var salesCatalogueResponse = await salesCatalogueService.PostProductIdentifiersAsync(productIdentifierRequest.ProductIdentifier.ToList());
+            ////can check for file size from salesCatalogueResponse.ResponseCode
+            var response = SetExchangeSetResponse(salesCatalogueResponse, false);
+            if (response.HttpstatusCode != HttpStatusCode.OK && response.HttpstatusCode != HttpStatusCode.NotModified)
             {
-                Href = @"http://fss.ukho.gov.uk/batch/7b4cdf10-adfa-4ed6-b2fe-d1543d8b7272"
-            };
-            LinkSetFileUri linkSetFileUri = new LinkSetFileUri()
-            {
-                Href = @"http://fss.ukho.gov.uk/batch/7b4cdf10-adfa-4ed6-b2fe-d1543d8b7272/files/exchangeset123.zip",
-            };
-            Links links = new Links()
+                return response;
+            }
+            //// FSS call for creating Batch and fill data for _links, exchangeSetUrlExpiryDateTime etc
+            response.ExchangeSetResponse.Links = new Links()
             {
-                ExchangeSetBatchStatusUri = linkSetBatchStatusUri,
-                ExchangeSetFileUri = linkSetFileUri
-            };
-            List<RequestedProductsNotInExchangeSet> lstRequestedProductsNotInExchangeSet = new List<RequestedProductsNotInExchangeSet>()
-            {
-                new RequestedProductsNotInExchangeSet()
-                {
-                    ProductName = "GB123456",
-                    Reason = "productWithdrawn"
-                },
-                new RequestedProductsNotInExchangeSet()
-                {
-                    ProductName = "GB123789",
-                    Reason = "invalidProduct"
-                }
+                ExchangeSetBatchStatusUri = new LinkSetBatchStatusUri { Href = "http://fss.ukho.gov.uk/batch/7b4cdf10-adfa-4ed6-b2fe-d1543d8b7272" },
+                ExchangeSetFileUri = new LinkSetFileUri { Href = "http://fss.ukho.gov.uk/batch/7b4cdf10-adfa-4ed6-b2fe-d1543d8b7272/files/exchangeset123.zip" }
```

Comment:
Is there a reason to hard code these Url's into the code or should they be moved into a configuration file?

Reply:
Yes this is been moved to common method and made configuration in PBI 2790 where it actually creates the Batch and based on that those links is generated

Reply:
Thanks